consensus/ConsensusFileGenerator.py

```diff
@@ -231,44 +229,53 @@ def generate_consensus_files(self):
         :param lab_classifications:
         :return: tuple with the names of the files (consensus_file, comments_file)
         """
-        comments_file_name = self.comments_table_file_name + '.csv'
-        consensus_file_name = self.consensus_table_file_name + '.csv'
+        comments_filename = f'{self.comments_table_file_name}.csv'
+        consensus_filename = f'{self.consensus_table_file_name}.csv'
+        blue_comments_filename = colored(comments_filename, 'blue')
+        blue_consensus_filename = colored(consensus_filename, 'blue')
 
-        print('\nWriting consensus table to [{}] and comments table to [{}]'.format(
-            colored(consensus_file_name, 'blue'),
-            colored(comments_file_name, 'blue')))
+        print(
+            f'\nWriting consensus table to [{blue_consensus_filename}] and comments table to [{blue_comments_filename}]')
 
         # Start progressbar
-        progress_bar = progressbar.ProgressBar(max_value=len(self.consensus_data))
+        progress_bar = progressbar.ProgressBar(max_value=len(self.consensus))
         progress = 0
 
-        # Initiate output files
-        consensus_file = open(consensus_file_name, 'w')
-        comments_file = open(comments_file_name, 'w')
+        # Create file content
+        consensus_file_content = ''
+        comments_file_content = ''
 
         # Create headers
-        labs = next(iter(self.lab_classifications.values()))
-        header = self.create_consensus_header(labs)
-        consensus_file.write(header)
-        comments_file.write('"id","comments"\n')
-
-        for variant_id in self.consensus_data:
-            variant_lab_classifications = self.lab_classifications[variant_id]
-            line = self._create_consensus_line(variant_id, self.consensus_data[variant_id], variant_lab_classifications,
-                                               labs)
-            consensus_file.write(line)
-            comment = '"{}","-"\n'.format(variant_id)
-            comments_file.write(comment)
-            progress += 1
-            progress_bar.update(progress)
+        consensus_file_content += self.create_consensus_header(self.labs)
+        comments_file_content += '"id","comments"\n'
 
-        consensus_file.close()
+        # Open output files
+        consensus_file = open(consensus_filename, 'w')
+        comments_file = open(comments_filename, 'w')
+
+        for i, variant_id in enumerate(self.consensus):
+            variant = self.consensus[variant_id]
+            consensus_file_content += self._create_consensus_line(variant_id, variant['consensus'],
+                                                                  variant['lab_classifications'], self.labs)
+            comments_file_content += f'"{variant_id}","-"\n'
+            if (i + 1) % 1000 == 0:
+                progress += 1000
+                progress_bar.update(progress)
+                consensus_file.write(consensus_file_content)
```

Comment:
If you have the file already opened, is there a reason not to write lines instantly instead of through progress? Would delete lines 269 and 270 in the progress to.

Reply:
Because it makes my code faster. In the previous version I wrote each line immediately after processing, but that's very inefficient. Then I wanted to add everything to a string and write it at once, but it took a lot of memory, making the code slower over time.

Reply:
That's a very fair reason, understood.